### ISSUE TITLE
Codebook vs Commit Loss: Mislabeled?

### ISSUE THREAD

**Issue by waynchi** [openai/jukebox#129]
Hi! I'm going through the codebase trying to understand the various losses that you use. The reconstruction loss is simple to understand, but i'm a little confused around the codebook loss and commit loss.

In the vqvae.py and bottleneck.py, there is something referred to as the "commit loss", but the line of code that generates it in bottleneck.py on line 172:
` commit_loss = t.norm(x_d.detach() - x) ** 2 / np.prod(x.shape)`
seems to match the codebook loss in the your paper. Is this mislabeled or am I missing something here?

Also, I can't seem to find the third main loss (whether it's commit or codebook loss) anywhere. Could you perhaps point me to where that would be? 

Thanks!

**Comment by MlWoo** [comment]
@waynchi  The paper explains it very clear. "To speed up training, the codebook loss of codebook instead uses EMA up- dates over the codebook variables."
